core/org.wso2.carbon.registry.core/src/main/java/org/wso2/carbon/registry/core/utils/RegistryUtils.java

```diff
@@ -185,18 +186,30 @@ public static InputStream getMemoryStream(InputStream inputStream) throws Regist
      * @return the unique identifier.
      */
     public static String getConnectionId(Connection connection) {
+        String connectionId = null;
         try {
             // The connection URL is unique enough to be used as an identifier since one thread
             // makes one connection to the given URL according to our model.
             DatabaseMetaData connectionMetaData = connection.getMetaData();
+            String productName = connectionMetaData.getDatabaseProductName();
             if (connectionMetaData != null) {
+                if (MY_SQL_PRODUCT_NAME.equals(productName)) {
+                    /*
+                     For MySQL getUserName() method executes 'SELECT USER()' query on DB via mysql connector
+                     causing a huge number of 'SELECT USER()' queries to be executed.
+                     Hence removing username when the DB in use is MySQL.
+                     */
+                    connectionId = connectionMetaData.getURL();
+                } else {
+                    connectionId = connectionMetaData.getUserName() + "@" + connectionMetaData.getURL();
+                }
                 return (connectionMetaData.getUserName() != null ? connectionMetaData.getUserName().split("@")[0] :
                         connectionMetaData.getUserName()) + "@" + connectionMetaData.getURL();
             }
-        } catch (SQLException ignore) {
-            log.debug("Failed to construct the connectionId ." + ignore.getMessage());
+        } catch (SQLException e) {
+            log.error("Failed to construct the connectionId.", e);
```

Comment:
Hi,
Thanks for the comment. This was not thrown because this fix is done for a patch and if this was thrown an API change[1] should be done. Added a public JIRA[2] to correct this properly in the future release. I'll send the PR for the JIRA. 

[1] - https://github.com/wso2/carbon4-kernel/blob/bdb47e3377ed932598fe473523c5eab4d174d0b7/core/org.wso2.carbon.registry.core/src/main/java/org/wso2/carbon/registry/core/utils/RegistryUtils.java#L188
[2] - https://wso2.org/jira/browse/CARBON-15224

Thanks